packages/astro/src/core/config/schema.ts

```diff
@@ -322,17 +325,31 @@ export const AstroConfigSchema = z.object({
 				routing: z
 					.object({
 						prefixDefaultLocale: z.boolean().default(false),
+						redirectToDefaultLocale: z.boolean().default(true),
 						strategy: z.enum(['pathname']).default('pathname'),
 					})
 					.default({})
+					.refine(
+						({ prefixDefaultLocale, redirectToDefaultLocale }) => {
+							return !(prefixDefaultLocale === false && redirectToDefaultLocale === false);
+						},
+						{
+							message:
+								'The option `i18n.redirectToDefaultLocale` has effects only when the `i18n.prefixDefaultLocale` is set to `true`. Remove the option `i18n.redirectToDefaultLocale`, or change the value of `i18n.prefixDefaultLocale` to `true`.',
```

Comment:
```suggestion
								'The option `i18n.redirectToDefaultLocale` is only useful when the `i18n.prefixDefaultLocale` is set to `true`. Remove the option `i18n.redirectToDefaultLocale`, or change its value to `true`.',
```

Maybe this is a bit more concise.